pkg/cli/iam_handle_test.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cli
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/abcxyz/access-on-demand/apis/v1alpha1"
+	"github.com/abcxyz/pkg/logging"
+	"github.com/abcxyz/pkg/testutil"
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestIAMHandleCommand(t *testing.T) {
+	t.Parallel()
+
+	// Set up IAM request file.
+	requestFileContentByName := map[string]string{
+		"valid.yaml": `
+policies:
+- resource: organizations/foo
+  bindings:
+  - members:
+    - user:test-org-userA@example.com
+    - user:test-org-userB@example.com
+    role: roles/cloudkms.cryptoOperator
+  - members:
+    - user:test-org-userA@example.com
+    - user:test-org-userB@example.com
+    role: roles/accessapproval.approver
+- resource: folders/bar
+  bindings:
+    - members:
+      - user:test-folder-user@example.com
+      role: roles/cloudkms.cryptoOperator
+- resource: projects/baz
+  bindings:
+    - members:
+      - user:test-project-user@example.com
+      role: roles/bigquery.dataViewer
+`,
+		"invalid.yaml": `bananas`,
+	}
+	dir := t.TempDir()
+	for name, content := range requestFileContentByName {
+		path := filepath.Join(dir, name)
+		if err := os.WriteFile(path, []byte(content), 0o600); err != nil {
+			t.Fatal(err)
+		}
+		t.Cleanup(func() {
+			if err := os.Remove(path); err != nil {
+				t.Fatal(err)
+			}
+		})
```

Comment:
Per https://pkg.go.dev/testing#T.TempDir, the dir will be auto removed after the test.